eth/tracers/native/prestate.go

```diff
@@ -229,7 +229,7 @@ func (t *prestateTracer) CaptureTxEnd(restGas uint64) {
 	// the new created contracts' prestate were empty, so delete them
 	for a := range t.created {
 		// the created contract maybe exists in statedb before the creating tx
-		if s := t.pre[a]; s.Balance.Cmp(big.NewInt(0)) == 0 && len(s.Storage) == 0 && len(s.Code) == 0 {
+		if s := t.pre[a]; s != nil && s.Balance.Cmp(big.NewInt(0)) == 0 && len(s.Storage) == 0 && len(s.Code) == 0 {
 			delete(t.pre, a)
 		}
```

Comment:
Let's try to make it more readable by not putting everything in the same clause: 
```golang
		if s := t.pre[a]; s != nil{
			if s.Balance.Sign() == 0 && len(s.Storage) == 0 && len(s.Code) == 0{
				delete(t.pre, a)
			}
		}
```

Reply:
thanks